scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketSource.java

```diff
@@ -453,20 +484,19 @@ public Predicate<V> getPredicate() {
     }
 
     public Coder<V> getCoder() {
-      return fileOperations.getCoder();
+      // Todo what if two input directories use a different Coder for the same type?
```

Comment:
For the same SpecificRecord type `T`, `ParquetFileOperations` returns a `Coder<GenericRecord>`, compared to `AvroFileOperations`, which returns a `Coder<T>`... while my unit test passed, I'm not totally comfortable with the ramifications of picking a coder at random 😬